app/views/Settings.js

```diff
@@ -63,12 +64,15 @@ export const SettingsScreen = ({ navigation }) => {
   const localeChanged = async locale => {
     setUserLocale(locale);
 
-    // If user picks manual lang, update and store setting
-    try {
-      await languages.changeLanguage(locale);
-      await SetStoreData(LANG_OVERRIDE, locale);
-    } catch (e) {
-      console.log('something went wrong in lang change', e);
+    if (languages.dir() === languages.dir(locale)) {
```

Comment:
only change language if it's the same dir?

Is that because elsewhere it refreshes the app if dir is changed?

The side effect is not obvious.

Reply:
Removed as now components aware of the direction